Replace setup.py with pyproject.toml

kpeez opened the pull request:
Replacing `setup.py` with `pyproject.toml` will permit the use of [poetry](https://python-poetry.org/) for package management. One advantage of using poetry to build rivuletpy and publish to pypi is that users can simply run `pip install rivuletpy` without having to worry about setting up the package dependencies themselves.

kpeez committed:
poetry setup

kpeez committed:
fixed python version

kpeez committed:
fixed scikit-image version

lsqshr commented:
Hey Kyle, thanks for the merge request. Good stuff! I was not familiar with poetry but happy to merge it.

Would you help check two things before merging:
(1) is the poetry.lock needed here?
(2) would you help check if anything in the readme needs to be updated accordingly?

Regards,
Siqi

kpeez committed:
build: fixed error building C extensions

- added build.py
- updated info in pyproject.toml

kpeez committed:
clean: VS Code auto-formatting

kpeez committed:
chore: remove poetry.lock file

kpeez committed:
docs: Added new installation instructions

kpeez committed:
chore: fixed typos

kpeez commented:
Hey Siqi,

1. I believe the lockfile is only recommended for deployed apps (see [this stack thread](https://stackoverflow.com/questions/61037557/should-i-commit-lock-file-changes-separately-what-should-i-write-for-the-commi), and the [official recommendation from the poetry devs](https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control)). I removed it and added it to the .gitignore file.
2. I updated the Installation instructions in the README to:

## Installation

### Setting up virtual environment

It is recommended to install rivulet in a virtual enviornment.

```bash
# create env and activate it
conda create -n riv
conda activate riv
# install pip and git
conda install pip git
```

### Install from PyPI

To install rivuletpy from PyPI simply activate your virtual environment and run:

```bash
pip install rivuletpy
```

### Install from GitHub

Optionally, you can use `pip` to install the latest version directly from GitHub:

```bash
pip install git+https://github.com/RivuletStudio/rivuletpy
```  

The second installation method should work after this merge. To update the PyPI version you'll need to generate an API key and add it to your poetry config. See ["Configure PyPI" in this post](https://mathspp.com/blog/how-to-create-a-python-package-in-2022#publish-the-real-package) for a quick overview on publishing packages with poetry.


Cheers,
Kyle

lsqshr commented:
Great! Thanks a lot Kyle! Will publish a new pypi version after the merge.
